rpi/gpio.go

```diff
@@ -12,21 +12,57 @@ import "C"
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/pkg/errors"
 	"go.viam.com/rdk/components/board"
 	rdkutils "go.viam.com/rdk/utils"
 	rpiutils "raspberry-pi/utils"
 )
 
+// GPIOConfig tracks what each pin is currently configured as
+type GPIOConfig int
+
+const (
+	GPIODefault   GPIOConfig = iota // GPIODefault is the default pin state, before we have modified the pin
+	GPIOInput                       // GPIOInput is the default pin state, before we have modified the pin
+	GPIOOutput                      // GPIOOutput is the default pin state, before we have modified the pin
+	GPIOPWM                         // GPIOPWM is the default pin state, before we have modified the pin
+	GPIOInterrupt                   // GPIOInterrupt is the default pin state, before we have modified the pin
```

Comment:
this is currently unused, but the thought would be to disallow any Set calls for a pin if the pin was configured as a digital interrupt. Can add this behavior if we want it now